src/glowCookies.js

```diff
@@ -235,55 +235,76 @@ class GlowCookies {
-      customScript: obj.customScript || undefined
-    }
+      this.tracking = {
+        AnalyticsCode: obj.analytics || undefined,
+        FacebookPixelCode: obj.facebookPixel || undefined,
+        HotjarTrackingCode: obj.hotjar || undefined,
+        customScript: obj.customScript || undefined
+      }
 
-    this.banner = {
-      description: obj.bannerDescription || lang.bannerDescription,
-      linkText: obj.bannerLinkText || lang.bannerLinkText,
-      link: obj.policyLink || '#link',
-      background: obj.bannerBackground || '#fff',
-      color: obj.bannerColor || '#1d2e38',
-      heading: obj.bannerHeading !== 'none' ? obj.bannerHeading || lang.bannerHeading : '',
-      acceptBtn: {
-        text: obj.acceptBtnText || lang.acceptBtnText,
-        background: obj.acceptBtnBackground || '#253b48',
-        color: obj.acceptBtnColor || '#fff'
-      },
-      rejectBtn: {
-        text: obj.rejectBtnText || lang.rejectBtnText,
-        background: obj.rejectBtnBackground || '#E8E8E8',
-        color: obj.rejectBtnColor || '#636363'
-      },
-      manageCookies: {
-        color: obj.manageColor || '#1d2e38',
-        background: obj.manageBackground || '#fff',
-        text: obj.manageText || lang.manageText,
+      this.banner = {
+        description: obj.bannerDescription || lang.bannerDescription,
+        linkText: obj.bannerLinkText || lang.bannerLinkText,
+        link: obj.policyLink || '#link',
+        background: obj.bannerBackground || '#fff',
+        color: obj.bannerColor || '#1d2e38',
+        heading: obj.bannerHeading !== 'none' ? obj.bannerHeading || lang.bannerHeading : '',
+        acceptBtn: {
+          text: obj.acceptBtnText || lang.acceptBtnText,
+          background: obj.acceptBtnBackground || '#253b48',
+          color: obj.acceptBtnColor || '#fff'
+        },
+        rejectBtn: {
+          text: obj.rejectBtnText || lang.rejectBtnText,
+          background: obj.rejectBtnBackground || '#E8E8E8',
+          color: obj.rejectBtnColor || '#636363'
+        },
+        manageCookies: {
+          color: obj.manageColor || '#1d2e38',
+          background: obj.manageBackground || '#fff',
+          text: obj.manageText || lang.manageText,
+        }
       }
-    }
 
-    // Draw banner
-    window.addEventListener('load', () => { this.render() })
+      // Draw banner
+      window.addEventListener('load', () => { this.render() })
+    }
   }
 }
 
 class LanguagesGC {
-  constructor(code) {
+  constructor(code, defaultLang) {
     this.init()
-    let lang = this.arrLang[code] || this.arrLang['en']
+    var lang = null;
+    if (code == "auto") {
+      var languages = [].concat(window.navigator.languages || [window.navigator.language || window.navigator.userLanguage]);
+      if (typeof defaultLang == "string")
+        languages.push(defaultLang);
+      languages.push('en');
+      for (let i = 0; i < languages.length; i++) {
+        // To-do: BCP47 to ISO-639 conversion
```

Comment:
Is it necessary to launch the functionality?

Reply:
It works even without, but it's preferable to implement. For example: in the languages array we can get things like `en-US`, `it-CH`, `pt-BR` and also `art-jbo` (wich is the same as `jbo`).